source/DataElement/AMDSFlatArray.cpp

```diff
@@ -718,38 +718,41 @@ bool AMDSFlatArray::resetTargetArrayAndReplaceData(AMDSFlatArray *targetArray) c
 }
 
 void AMDSFlatArray::resizeType(AMDSDataTypeDefinitions::DataType dataType, quint32 size){
+	if( (dataType_ == dataType) && (size == this->size()) )
+		return;
+
 	dataType_ = dataType;
```

Comment:
That's a good point. I think it's fine (that is, I named this function poorly). The only way to access this publicly is through the clearAndReset() function ... thus, the clear is already done.